routes/index.js

```diff
@@ -196,35 +196,140 @@ module.exports = function (passport) {
       dbAPI.getStations()
-
     ])
+
       .then(
-        ([allUsers, allStations]) => {
+        ([usersByDate, privUsers, allStations]) => {
+          if (!usersByDate) {
+            const err = "Interal error: Unable to get users."
+            req.flash('error', err);
+            res.redirect('/userManagement');
+            res.status(500).send(err + "via the /userManagement (post) route.");
+          }
+
+          /*Filter by most specific items first to limit number of comparisons.
+          *Also, if user didn't provide a particular input, no comparisons happen at that step.
+          *Must check all provided input, however; cannot return match prematurely as later filter
+          *may weed it out.*/
+          var stationMatchesArray = new Array();
+          var badgeFilter = req.body.badgeInput
+                              ? usersByDate.filter( x => 
+                                  x.badge.indexOf(req.body.badgeInput) > -1)
+                              : usersByDate;
+          var nameFilter = req.body.nameInput
+                              ? badgeFilter.filter( x => 
+                                  (x.first + " " + x.last)
+                                    .toUpperCase()
+                                    .indexOf(req.body.nameInput.toUpperCase()) > -1)
+                              : badgeFilter;
+          var statusFilter = undefined; //specified below; too complex for ternery operator
+          
           var data = {
-            users: allUsers,
             stations: allStations,
             authenticated: true
+            //users get added below
+          }
+
+          if ( //User did not specify search by status or explicity chose "All" statuses
+            (req.body.statusSelector && req.body.statusSelector === "All")
+            || (req.body.statusSelector === undefined)
+          ) {
+            statusFilter = nameFilter;
+          } else {
+            //User specified a particular status by which we should search
+            statusFilter = nameFilter.filter( x => x.status === req.body.statusSelector);
           }
 
+
+          if ( //User did not specify search by station, or explicitly chose "All" stations
+            (req.body.stationSelector && req.body.stationSelector === "All")
+            || (req.body.stationSelector === undefined) 
+          ) {
+            data.users = statusFilter;
+          } else {
+            //If user specified a particular station, iterate over getPrivilegedStationUsers results
+            privUsers.forEach( function (privElement) {
+              statusFilter.forEach( function (statusElement) {
+                if (privElement.badge == statusElement.badge) {
+                  stationMatchesArray.push(privElement);
+                }
+              });
+            });
+
+            data.users = stationMatchesArray;
+          }
+          
           res.render('userManagement.njk', data);
```

Comment:
I think the problem for the Confirm Form Re submission is still there is because you are trying to render the userManagement page in the post. Essentailly what should happen is that this post returns a JSON object with all the matched users according to given search criteria.